core/src/test/kotlin/download/PluginDownloadTest.kt

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.admin.download
+
+import com.epam.drill.admin.plugins.*
+import kotlinx.coroutines.*
+import java.io.*
+import kotlin.test.*
+
+class PluginDownloadTest {
+
+    private lateinit var pluginStorageDir: File
+
+    @BeforeTest
+    fun before() {
+        pluginStorageDir = File("build/tmp/test/plugin/storage/").also {
+            it.mkdirs()
+        }
+    }
+
+    @AfterTest
+    fun after() {
+        pluginStorageDir.deleteRecursively()
+    }
+
+    @Test
+    fun `should successfully download t2c plugin`() = runBlocking {
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("test2code"))
+        assert(pluginStorageDir.listFiles()?.size == 1)
+    }
+
+    @Test
+    fun `should be no exceptions when downloading the unknown plugin`() = runBlocking {
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("unknown plugin"))
+        assert(pluginStorageDir.listFiles()?.size == 0)
+    }
+
+    @Test
+    fun `should successfully download only available plugins`() = runBlocking {
+        val loaderService = ArtifactoryPluginLoader(Artifactory.GITHUB, pluginStorageDir, false)
+        loaderService.loadPlugins(listOf("state-watcher", "unknown plugin"))
+        assert(pluginStorageDir.listFiles()?.size == 1)
```

Comment:
Is it ok that tests spend a quota of downloading?

Reply:
Well, in our case, when checks very often fall, this is not normal, I think it is worth turning off these tests until we fix the integration tests.